components/sup/src/manager/file_watcher.rs

```diff
@@ -1,3464 +1,2662 @@
+            fs::remove(&descendent_file)?;
+            fs::remove(&descendent_dir)?;
 
-                Some(dir_path)
-            } else {
-                None
-            };
-            // any watch drop means that we don't see the real file anymore
-            self.real_file = None;
-            if let Some(prev) = common.prev {
-                // The `prev` patch may not be watched anymore when we
-                // stop watching a bunch of files in a batch (like
-                // when we move the watched directory away, so we stop
-                // watching everything inside it).
-                if let Some(watched_prev) = self.get_mut_watched_file(&prev) {
-                    watched_prev.get_mut_common().next = None;
-                }
-            } else {
-                error!("Prev path inconsistency in drop_watch, expect strange results");
-            }
-            dir_to_unwatch
-        } else {
-            error!("Paths inconsistency in drop_watch, expect strange results");
-            None
+            Ok(())
         }
-    }
 
-    fn symlink_loop(&mut self,
-                    path: &Path,
-                    path_rest: &VecDeque<OsString>,
-                    new_path: &Path,
-                    new_path_rest: &VecDeque<OsString>)
-                    -> bool {
-        let mut merged_path_rest = new_path_rest.clone();
-        merged_path_rest.extend(path_rest.iter().cloned());
-        let mut merged_path = new_path.to_path_buf();
-        merged_path.extend(merged_path_rest);
-        match self.symlink_loop_catcher.entry(path.to_path_buf()) {
-            Entry::Occupied(o) => *o.get() == merged_path,
-            Entry::Vacant(v) => {
-                v.insert(merged_path);
-                false
-            }
+        // what to test
+        // (1a) existing file with extension
+        // (1b) existing file without extension
+        // (2) existent directory / is_dir() violated
+        // (3a) ends with a path separator violation
+        // (3b) ends with a path separator acceptance (file without extension)
+        // (4a) parsable enclosing directory that exists
+        // (4b) parsable enclosing directory that does not exist
+        // (5) valid path but neither file nor directory exist
+
+        #[test]
+        fn preclude_directories() -> Result<()> {
+            // crate::logger::init()
+
+            // TESTING (3a)
+            // This test case helped tease out something useful.  We have an expectation that input
+            // is platform appropriate.  The original version of this test was a string with the
+            // unix style foreslashes but that failed on Windows due to the fore/back slash
+            // difference and since this was the only place it bit this felt OK.
+            #[cfg(not(windows))]
+            let s = "/path/ending/in/file/separator/";
+            #[cfg(windows)]
+            let s = r"\\path\ending\in\file\separator\";
```

Comment:
keep in mind that forward slashes are legal on windows. I'm not sure if you create a `Path`  on windows if it will internally transform the forward slashed to backslashes. If not then you will need to check for both.